py/desispec/scripts/daily_processing.py

```diff
@@ -355,6 +355,8 @@ def daily_processing_manager(specprod=None, exp_table_path=None, proc_table_path
                 # If done with science exposures for a tile and use_tilenight==True, use
                 # submit_tilenight_and_redshifts, otherwise use checkfor_and_submit_joint_job
                 if use_tilenight and lasttype == 'science' and len(sciences)>0:
+                    if dry_run_level < 3:
+                        write_tables([etable], tablenames=[exp_table_pathname])                    
```

Comment:
This writes the latest exposures out to disk before trying to submit processing. In the event that the code crashes while trying to submit a job for processing, (e.g. when Slurm doesn't respond repeatedly), this will create the same type of issue we have previously seen. 

We can either move this code block to below the `submit_tilenight_and_redshifts()` call (just above the current `else` statement on line 370, or we can write the etable out when we write out the ptable on line 384. That will add an extra write for non-tilenight exposure tables, but it will be harmless.

Reply:
@akremin just to clarify, this writes the exposure(s) in the tile about to be processed to disk, but not the latest exposure, since the row corresponding to the latest exposure is added to the exposure table below around line 391.

It's true that if something goes wrong in `submit_tilenight_and_redshifts()` such that the job is not submitted and the job crashes, then the exposures can end up in the exposure table, and thus would never be processed. I ended up placing the `write_table` call there because ztile jobs were missing the last exposure and thus incorrect (the tilenights were correct, however), and thus I assumed something about the ztile job creation was reading the exposure table. But I could have been mistaken that placing the `write_table` call before the `submit_tilenight_and_redshifts()` call actually fixed the missing ztile exposure error.

Do you think it's possible that somewhere in the ztile job creation that the exposure table is read in? I took a quick look and did not see it anywhere -- it seems to exclusively depend on the processing table passed in to `submit_tilenight_and_redshifts()` -- but it's complex enough that I could have just missed it.

In the meantime I will try moving the `write_table` call to just after `submit_tilenight_and_redshifts()` and see if that doesn't cause the ztile processing error to reappear in my mock observation tests.

Reply:
Sounds good, thanks. I will have a look at the two options you've suggested.